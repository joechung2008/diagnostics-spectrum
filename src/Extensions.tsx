import { ActionButton } from "@adobe/react-spectrum";
import { useMemo } from "react";
import { byKey, isExtensionInfo, toNavLink } from "./utils";

import styles from "./Extensions.module.css";

const Extensions: React.FC<ExtensionsProps> = ({ extensions, onLinkClick }) => {
  const links = useMemo(
    () =>
      Object.values(extensions)
        .filter(isExtensionInfo)
        .map(toNavLink)
        .sort(byKey),
    [extensions]
  );

  return (
    <nav className={styles["extensions-root"]} aria-label="Extensions">
      {links.map((link) => (
        <ActionButton
          key={link.key}
          UNSAFE_className={styles.navButton}
          onPress={() => onLinkClick?.(undefined, link)}
        >
          {link.name}
        </ActionButton>
      ))}
    </nav>
  );
};

export default Extensions;
