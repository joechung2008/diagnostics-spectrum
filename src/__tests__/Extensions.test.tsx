import { describe, expect, it } from "vitest";
import Extensions from "../Extensions";
import { render } from "./test-utils";

describe("Extensions", () => {
  it("renders correctly with extensions", () => {
    const extensions = {
      ext1: { extensionName: "extension1", config: { key: "value" } },
      ext2: { extensionName: "extension2" },
    };
    const { container } = render(
      <Extensions extensions={extensions} onLinkClick={() => {}} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with empty extensions", () => {
    const extensions = {};
    const { container } = render(
      <Extensions extensions={extensions} onLinkClick={() => {}} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
