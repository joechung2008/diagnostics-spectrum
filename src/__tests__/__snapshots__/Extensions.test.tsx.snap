// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Extensions > renders correctly with empty extensions 1`] = `
<div
  class="_t8qIa_spectrum kDKRXa_spectrum kDKRXa_i18nFontFamily YqfL3a_spectrum--light _1DrGeG_spectrum--large XhWg9q_spectrum XhWg9q_spectrum--dark XhWg9q_spectrum--darkest XhWg9q_spectrum--large XhWg9q_spectrum--light XhWg9q_spectrum--lightest XhWg9q_spectrum--medium"
  dir="ltr"
  lang="en-US"
  style="isolation: isolate; color-scheme: light;"
>
  <nav
    aria-label="Extensions"
    class="_extensions-root_86ca92"
  />
</div>
`;

exports[`Extensions > renders correctly with extensions 1`] = `
<div
  class="_t8qIa_spectrum kDKRXa_spectrum kDKRXa_i18nFontFamily YqfL3a_spectrum--light _1DrGeG_spectrum--large XhWg9q_spectrum XhWg9q_spectrum--dark XhWg9q_spectrum--darkest XhWg9q_spectrum--large XhWg9q_spectrum--light XhWg9q_spectrum--lightest XhWg9q_spectrum--medium"
  dir="ltr"
  lang="en-US"
  style="isolation: isolate; color-scheme: light;"
>
  <nav
    aria-label="Extensions"
    class="_extensions-root_86ca92"
  >
    <button
      class="o7Xu8a_spectrum-ActionButton o7Xu8a_spectrum-BaseButton o7Xu8a_i18nFontFamily o7Xu8a_spectrum-FocusRing o7Xu8a_spectrum-FocusRing-ring _navButton_86ca92"
      data-react-aria-pressable="true"
      tabindex="0"
      type="button"
    >
      <span
        class="o7Xu8a_spectrum-ActionButton-label"
        role="none"
      >
        extension1
      </span>
    </button>
    <button
      class="o7Xu8a_spectrum-ActionButton o7Xu8a_spectrum-BaseButton o7Xu8a_i18nFontFamily o7Xu8a_spectrum-FocusRing o7Xu8a_spectrum-FocusRing-ring _navButton_86ca92"
      data-react-aria-pressable="true"
      tabindex="0"
      type="button"
    >
      <span
        class="o7Xu8a_spectrum-ActionButton-label"
        role="none"
      >
        extension2
      </span>
    </button>
  </nav>
</div>
`;
